client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -145,23 +145,27 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionFactory
     @Override
     public <T> EventStreamWriter<T> createEventWriter(String writerId, String streamName, Serializer<T> s,
                                                       EventWriterConfig config) {
-        log.info("Creating writer for stream: {} with configuration: {}", streamName, config);
+        log.info("Creating writer: {} for stream: {} with configuration: {}", writerId, streamName, config);
         Stream stream = new StreamImpl(scope, streamName);
         ThreadPoolExecutor retransmitPool = ExecutorServiceHelpers.getShrinkingExecutor(1, 100, "ScalingRetransmition-"
                 + stream.getScopedName());
         return new EventStreamWriterImpl<T>(stream, writerId, controller, outFactory, s, config, retransmitPool, connectionFactory.getInternalExecutor());
     }
 
     @Override
-    @SuppressWarnings("deprecation")
     public <T> TransactionalEventStreamWriter<T> createTransactionalEventWriter(String writerId, String streamName,
                                                                                 Serializer<T> s,
                                                                                 EventWriterConfig config) {
-        log.info("Creating transactional writer for stream: {} with configuration: {}", streamName, config);
+        log.info("Creating transactional writer:{} for stream: {} with configuration: {}", writerId, streamName, config);
         Stream stream = new StreamImpl(scope, streamName);
         return new TransactionalEventStreamWriterImpl<T>(stream, writerId, controller, outFactory, s, config, connectionFactory.getInternalExecutor());
     }
 
+    @Override
+    public <T> TransactionalEventStreamWriter<T> createTransactionalEventWriter(String streamName, Serializer<T> s, EventWriterConfig config) {
+        return createTransactionalEventWriter(UUID.randomUUID().toString(), streamName, s, config);
```

Comment:
My only concern here is that `StreamConfiguration.timestampAggregationTimeout` needs to be set to a reasonable value here, otherwise an application unaware might endup holding up watermarks.

I suppose given that the default is sane, it's ok to let users configure something badly. But I think we may want to make a note in the Javadoc, perhaps recommending they set `automaticallyNoteTime(true)` on the `EventWriterConfig`

Reply:
Updated the java doc to indicate that writer id is autogenerated.
(io.pravega.client.stream.Transaction#commit(long) will be used to note the time and the default value of `timestampAggregationTimeout` is 0)